modules/compute/vm-instance/main.tf

```diff
@@ -45,6 +45,18 @@ locals {
     "ENABLE"  = "TRUE"
   }
   enable_oslogin = var.enable_oslogin == "INHERIT" ? {} : { enable-oslogin = lookup(local.oslogin_api_values, var.enable_oslogin, "") }
+
+  machine_vals            = split("-", var.machine_type)
+  machine_family          = local.machine_vals[0]
+  machine_not_shared_core = length(local.machine_vals) > 2
+  machine_vcpus           = try(parseint(local.machine_vals[2], 10), 1)
+
+  smt_capable_family = !contains(["t2d"], local.machine_family)
+  smt_capable_vcpu   = local.machine_vcpus >= 2
+
+  smt_capable = local.smt_capable_family && local.smt_capable_vcpu && local.machine_not_shared_core
+  smt_enable  = var.threads_per_core != null && (var.threads_per_core == 0 && local.smt_capable || try(var.threads_per_core >= 1, false))
+  smt_value   = var.threads_per_core == 2 ? 2 : 1
```

Comment:
nit: I prefer `threads_per_core` as the variable name here as it is more explicit. Basically, what is a _"Simultaneous Multithreading value"_. Terraform should be able to handle having a var and local of the same name. So you would have: 

Line 59
```
threads_per_core   = var.threads_per_core == 2 ? 2 : 1
```

Line 148
```
threads_per_core = local.threads_per_core
```

Reply:
Changed. Thanks!